static/src/javascripts/projects/common/modules/commercial/hosted-video.js

```diff
@@ -85,101 +87,105 @@ define([
     }
 
     function init() {
-        require(['bootstraps/enhanced/media/main'], function () {
-            require(['bootstraps/enhanced/media/video-player'], function(videojs){
-                var $videoEl = $('.vjs-hosted__video');
-
-                if (!$videoEl.length) {
-                    return;
-                }
-
-                player = videojs($videoEl.get(0), videojsOptions());
-                player.guMediaType = 'video';
-                videojs.plugin('fullscreener', fullscreener);
-
-                player.ready(function () {
-                    var vol;
-                    var duration = parseInt(this.duration(), 10);
-                    var $hostedNext = $('.js-hosted-next-autoplay');
-                    initLoadingSpinner(player);
-                    upgradeVideoPlayerAccessibility(player);
-
-                    // unglitching the volume on first load
-                    vol = player.volume();
-                    if (vol) {
-                        player.volume(0);
-                        player.volume(vol);
+        return new Promise(function (resolve) {
+            require(['bootstraps/enhanced/media/main'], function () {
+                require(['bootstraps/enhanced/media/video-player'], function (videojs) {
```

Comment:
Man this function is looooooooooong. Might be worth abstracting a little in another PR 😮